src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -38,5 +49,58 @@ public boolean existsByName(String identifier) {
   } // TODO: Ensure this successfully searches by email
 
   @Override
-  public void save(User user) {}
+  public String save(User user) {
+    String firebaseResponse = firebaseSignup(user.getEmail(), user.getPassword());
+    return firebaseResponse;  // To deal with error handling in the interactors
+  }
+
+  public String firebaseSignup(String email, String password) {
+    try {
+      URL url = new URL("https://identitytoolkit.googleapis.com/v1/accounts:signUp?key=AIzaSyBuJk14Gljk-chdN_9YVywSKnf38ttwUVg");
+      HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+      conn.setRequestMethod("POST");
+      conn.setRequestProperty("Content-Type", "application/json; utf-8");
+      conn.setRequestProperty("Accept", "application/json");
+      conn.setDoOutput(true);
```

Comment:
This can be handled by OkHTTP.
Using HttpURLConnection makes the code more complex without giving us any advantages in return.